formulation descriptor and (some) predictors (#324)
CitrineInformatics/citrine-python: docs/source/workflows/data_sources.rst

        training_data = data_source
    )

An optional list of identifiers can be specified.
Identifiers uniquely identify a row and are used in the context of simple mixtures to link from an ingredient to the its properties.

> Each identifier must correspond to a column header name

Reviewer: ```suggestion
Each identifier must correspond to a column header name.
```

No two rows within this column can contain the same value.

> Identifier columns may overlap with the keys defined in the mapping from column header names to descriptors if a column should be parsed as data and used as an identifier.

Reviewer: ```suggestion
If a column should be parsed as data and used as an identifier, identifier columns may overlap with the keys defined in the mapping from column header names to descriptors.
```

Reviewer: Rearranging so that a comma can help differentiate ideas in reading.

Identifiers are required in two circumstances.
These circumstances are only relevant if CSV data source represents simple mixture data.

1. Ingredient properties are featurized using a :class:`~citrine.informatics.predictors.GeneralizedMeanPropertyPredictor`.
   In this case, the link from identifier to row is used to compute mean ingredient property values.

> 2. Simple mixtures are boiled down to recipes that contain only atomic ingredients using a :class:`~citrine.informatics.predictors.SimpleMixturePredictor`.

Reviewer: ```suggestion
2. Simple mixtures that contain mixtures are simplified to recipes that contain only atomic ingredients using a :class:`~citrine.informatics.predictors.SimpleMixturePredictor`.
```

>    In this case, links from each mixture's ingredients to its row (which may also be a mixture) are used to to recursively crawl hierarchical blends of blends and construct a recipe that contains only leaf ingredients.

Reviewer: ```suggestion
   In this case, links from each mixture's ingredients to its row (which may also be a mixture) are used to recursively crawl hierarchical blends of blends and construct a recipe that contains only leaf ingredients.
```

> Note, to build a formulation from a CSV data source an :class:`~citrine.informatics.predictors.IngredientsToSimpleMixturePredictor` must be present in the workflow.

Reviewer: ```suggestion
Note: to build a formulation from a CSV data source an :class:`~citrine.informatics.predictors.IngredientsToSimpleMixturePredictor` must be present in the workflow.
```
